turbo/transactions/tracing.go

```diff
@@ -70,11 +72,11 @@ func ComputeTxEnv(ctx context.Context, engine consensus.EngineReader, block *typ
 
 	blockContext := core.NewEVMBlockContext(header, core.GetHashFn(header, getHeader), engine, nil, cfg)
 
-	if txIndex == len(block.Transactions()) {
-		// tx is a state sync transaction
-		return nil, blockContext, evmtypes.TxContext{}, statedb, nil, nil
-	}
```

Comment:
seems this `if` is lost

Reply:
This was added in #12316 but is no longer needed with these changes.